config/rbac/role.yaml

```diff
@@ -39,6 +38,7 @@ rules:
   resources:
   - machines
   - machines/status
+  - machinesets
```

Comment:
In order to know when an affinity group needs deletion, the machine controller needs to watch for the deletion of its management owner.

In the case of workload nodes, that's the machineset pointed to by the CAPI machine that owns the CAPC machine.

Adding this permission allows the machine controller to list machinesets to check for deletion thereof.